### PR TITLE
FV-377 Keine Zähldaten bei 13h-Zählung und Spitzenstundenauswahl

### PR DESCRIPTION
**Pull request by boal**
# Description

Anpassung der Spitzenstundenberechnung bezüglich 13h-Zählung sowie der Einzelberechnung der Spitzenstunde für den Belastungsplan.

# Issue References

FV-377
 
# Definition of Done

- [x] Acceptance criteria are met
- [x] Testing is done (unit-tests, DEV-environment)
- [x] Build/Test workflow has successfully finished
- [x] Release notes are complemented
- [x] Documentation is complemented (operator manual, system specification, etc.)

## Release Notes

* **Refactor**
  * Enhanced peak hour calculation handling to better manage movement relationships in traffic analysis.
  * Improved internal organization of peak hour computation methods with refined grouping by movement type.

* **Tests**
  * Expanded test coverage for peak hour calculation methods and movement relationship assignments.
  * Added tests validating edge cases in peak hour processing.

### CODE REVIEW

File: src/main/java/de/muenchen/dave/domain/enums/Zeitblock.java [hunk 1]
```diff
@@ -462,4 +463,41 @@ public enum Zeitblock implements Serializable {
...
+     * </ul>
+     *
+     * Wenn {@code zeitblockToCompare} {@code null} ist, wird {@code false} zurückgegeben.
+     *
+     * @param zeitblockToCompare der Zeitblock, in dem geprüft werden soll, ob sich dieser Zeitblock
+     *            befindet
+     * @return {@code true} wenn dieser Zeitblock vollständig innerhalb von {@code zeitblockToCompare}
+     *         liegt, sonst {@code false}
+     */
+    public boolean isThisZeitblockWithinZeitblockToCompare(final Zeitblock zeitblockToCompare) {
```
Comment (sfi2022) [inline src/main/java/de/muenchen/dave/domain/enums/Zeitblock.java]: Lässt sich das auch mit weniger Zeilen Ausdrücken? 
Z.Bsp.:

 public boolean isThisZeitblockWithinZeitblockToCompar(final Zeitblock zeitblockToCompare) {
        return zeitblockToCompare != null
                && !getStart().isBefore(zeitblockToCompare.getStart())
                && !getEnd().isAfter(zeitblockToCompare.getEnd());
    }

**Review by sfi2022 — COMMENTED** [review]
Ich habe eine Anmerkung die du umsetzen kannst.
Die ganze Magie liegt in den Lambda Ausdrücken zur Berechnung - was würden wir nur ohne machen :-) 
Die Testfälle sind hoffentlich umfassend.

File: src/main/java/de/muenchen/dave/domain/enums/Zeitblock.java (hunk 1 above)
Comment (boal) [reply src/main/java/de/muenchen/dave/domain/enums/Zeitblock.java]: Ja, das hatte ich sogar im ersten Wurf.

Jedoch hatte ich dann Probleme dias zu lesen, nachdem ich einen kleinen Bug eingebaut hatte. Somit habe ich es auseinandergezogen um die einzelnen Schritte im Code zu erkennen.

**Review by sfi2022 — APPROVED** [review]

**Review by sfi2022 — APPROVED** [review]
